ext/stringio/stringio.c

```diff
@@ -1523,6 +1523,24 @@ strio_size(VALUE self)
 
 /*
  * call-seq:
+ *   strio.empty? -> true or false
+ *
+ * Returns the true when there is no content.
+ */
+static VALUE
+strio_empty(VALUE self)
+{
+    VALUE string = StringIO(self)->string;
+    if (NIL_P(string)) {
+      rb_raise(rb_eIOError, "not opened");
+    }
+    long len = RSTRING_LEN(string);
+    if (len == 0) { return Qtrue; }
```

Comment:
Since `len` is not used anywhere else but that `if`, can this be equivalent?
```c
if (RSTRING_LEN(string) == 0) { return Qtrue; }
```

Reply:
Good point!